mindmeld/components/entity_resolver.py

```diff
@@ -1220,50 +1490,56 @@ def __init__(self, app_path, **kwargs):
                         is average of all whitelist's (including above alterations) encodings.
                     scores_normalizer: a normalizer that normalizes computed similarity scores
                         allowed choices are "min_max_scaler", "standard_scaler"
-                    cache_path: a predetermined cache path to dump embeddings
                     batch_size: can be set based on machine capabilities and RAM
         """
-        super().__init__(app_path, **kwargs)
+        super().__init__(app_path, entity_type, **kwargs)
 
         er_config = kwargs.pop("config", {})
         settings = er_config.get("model_settings", {})
         self._aug_lower_case = settings.get("augment_lower_case", False)
         self._aug_title_case = settings.get("augment_title_case", False)
         self._aug_normalized = settings.get("augment_normalized", False)
         self._aug_avg_syn_embs = settings.get("augment_average_synonyms_embeddings", True)
+        self._normalize_aliases = False
+
+        self._embedder_cache_path = kwargs.get("embedder_cache_path")
 
-        cache_path = settings.get("cache_path")
-        if not cache_path:
-            hashid = f"{self.__class__.__name__}$synonym_{self.type}"
-            cache_path = get_entity_resolver_cache_file_path(app_path, hashid)
-            settings.update({"cache_path": cache_path})
-            er_config.update({"model_settings": settings})
+        # No `app_path` specified because we now have dump and load methods that take a caching path
+        # as argument. Previously, we just dumped and loaded from a default cache_path ascertained
+        # by that embedder model itself (through model_id and app_poth) i.e. one path for all usages
+        # of that particular embedder model. However, with the introduction of dump and load method
+        # to resolvers, one-path-for-all is no longer an optimal solution and is deprecated.
+        # Note that we do not specify any `cache_path` key field as well in the er_config here!
+        self._embedder_model = create_embedder_model(app_path=None, config=er_config)
```

Comment:
just wonder shouldn't we be initializing embedder model based on the ER config in `load()`?

Reply:
Thanks for noticing that, @mhuang2173 . Had to make few changes in entity_resolver.py to account for that. Please take a look now.

At the time of your comment, the QA module actually had the resolver config settings stored and hence it was not leading to any sort of bug. But I realized that could be a problem for embedder resolver when used outside the context of question answerers. Hence, I modified some code related to the way the config is saved in the resolvers and it should resolve this issue.